src/main/java/seedu/omnitravel/travelactivitytypes/TravelActivityList.java

```diff
@@ -65,24 +65,41 @@ public void listTravelActivities(){
     public int getNoOfTravelActivities(){
         return noOfActivities;
     }
+
     /**
      * Removes travel activity from the travel activity list
-     * @param activityNumber The travel activity number on the list
+     * @param activityNumber The travel activity index number or description on the list
      */
-    public void removeTravelActivity(int activityNumber) throws OmniException {
-        assert activityNumber != 0  :"There is not activities in the list";
-        if(activityNumber > travelActivities.size() || (activityNumber <= 0)){
-            throw new OmniException("Travel activity cannot be found!");
+    public void removeTravelActivity(String activityNumber) throws OmniException {
+        try {
+            assert Integer.parseInt(activityNumber) != 0 : "There is not activities in the list";
+            int indexOfActivity = Integer.parseInt(activityNumber) - 1;
+            int initialListSize = noOfActivities;
+            TravelActivity removedActivity = travelActivities.get(indexOfActivity);
+            travelActivities.remove(indexOfActivity);
+            System.out.println("I have removed this activity:");
+            System.out.println(removedActivity);
+            noOfActivities -= 1;
+            int newSize = noOfActivities;
+            assert newSize == initialListSize - 1 : "There is an error with list size!";
+        } catch (NumberFormatException e) {

```

Comment:
You should try to apply SLAP so that the code can be shorter and more readable. Currently the method is slightly too long.